Feature: save and load packets store

rpl opened the pull request:
This pull request (which will fix #22) adds a new File menu in the packets panel toolbar, the related save/load methods in the actions singleton and serialize/deserialize methods in the PacketsStore.
It renders a bootstrap based warning alert (auto dismissed within 2 seconds) on serializing/deserializing errors.

The two commits contain the karma unit tests which ensure the presence of the new packets panel toolbar menu and ensure the load/save actions are called when the user clicks on the corresponding menu items:

- https://travis-ci.org/rpl/rdp-inspector/builds/62935280#L287-L294

Follows a screencast of the feature:

![rdp-inspector-saveload-packets](https://cloud.githubusercontent.com/assets/11484/7671826/1477f558-fcde-11e4-84af-95b1227f6136.gif)

rpl commented:
@janodvarko I worked on fixing #22 before closing #35 because having dumps of the packets store will help to create the specs for the React components, and I wanted to test drive the karma suite introduced with #35 by developing this feature using a TDD workflow.

janodvarko commented:
Awesome!

* I have merged #35
* See my inline comments

Honza

janodvarko commented:
One more thing, when I apply the patch (the two related commits) the vertical splitter for the side bar is gone.

Honza

rpl commented:
@janodvarko I've rebased the pull request on the updated master, addressed the inline review comments (they are not visible anymore on this pull request because of the rebasing) in a bunch of new commits and fixed the regression on the vertical splitter.

janodvarko commented:
Works great, Luca!

Honza

janodvarko commented:
@rpl: are you planning some other work on the inspector atm? Or should I release new version and ask for review on AMO?

Honza

rpl commented:
@janodvarko I took a look to the changes merged from the last tag (https://github.com/firebug/rdp-inspector/compare/rdpinspector-1.0.3...master) and it seems that we've added a couple of features, there are at least the following three:

- the restored "Actors panel"
- auto-completion of the "to" attribute in the "Send Packet" sidebar
- the Save/Load of the Packets panel status)

I would like to take a look to the issues you mentioned in the comment https://github.com/firebug/rdp-inspector/pull/34#issuecomment-98681032

but if those issues are not blockers, we can prepare to tag and release this version on AMO and schedule the issues on the next one (I'm going to take a look on them as soon as I reach the tree-editor-view component with the unit testing).

janodvarko commented:
Sounds good, release done, https://github.com/firebug/rdp-inspector/releases/tag/rdpinspector-1.0.4
(also uploaded on AMO and waiting for review)

Honza
